e2e/nomostest/config_sync.go

```diff
@@ -699,8 +683,10 @@ func RootSyncObjectV1Alpha1FromRootRepo(nt *NT, name string) *v1alpha1.RootSync
 	repoURL := nt.GitProvider.SyncURL(repo.RemoteRepoName)
 	sourceFormat := repo.Format
 	rs := RootSyncObjectV1Alpha1(name, repoURL, sourceFormat)
-	if nt.DefaultReconcileTimeout != 0 {
-		rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(nt.DefaultReconcileTimeout)
+	if nt.DefaultReconcileTimeout != nil {
+		rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*nt.DefaultReconcileTimeout)
+	} else if rs.Spec.Override != nil {
+		rs.Spec.Override.ReconcileTimeout = nil
```

Comment:
I feel like this might be better form if we get rid of the `rs.Spec.Override != nil` check and do:
```golang
	} else {
		rs.Spec.SafeOverride().ReconcileTimeout = nil
	}
```
WDYT?

Reply:
I did it this way to avoid populating the override if I don't have to.
The way it is now, if override is nil o missing, it doesn't replace that with `override = { }`.

Reply:
For now, since the default is 1m, that case will never happen. but if we changed the default to nil, it would.